lib/shoryuken/options.rb

```diff
@@ -159,9 +159,15 @@ def active_job_queue_name_prefixing?
       @active_job_queue_name_prefixing
     end
 
+    def inline_executor?
+      worker_executor == Worker::InlineExecutor
+    end
+
     private
 
     def default_server_middleware
+      return Middleware::Chain.new if inline_executor?
```

Comment:
I'm wondering if we should create an `inline_middleware` and `configure_inline` instead of patching `server`. 

cc/ @cjlarose

Reply:
Yeah checking for `worker_executor == Worker::InlineExecutor` is a definite code smell: it's a violation of the tell-don't-ask principle.

I'd like to know more about the motivation for this change: it's clear that this implementation is trying to make it so that when using the `InlineExecutor`, worker-specific middleware is executed, but we don't execute any of the `default_server_middleware`. I could see that behavior as being unexpected for some users--the least surprising options are either

1) Just execute _all_ server middleware when using the InlineExecutor or
2) Make it clear in the API which middleware is executed when using different executors

I _think_ @phstc is suggesting something like

```ruby
Shoryuken.configure_inline do |config|
  config.inline_middleware do |chain|
    ...
  end
end
```

which mirrors

```ruby
Shoryuken.configure_server do |config|
  config.server_middleware do |chain|
  end
end
```

But I think if the main motivation for changing the default middleware stack is for testing purposes, then this is more simply delegated to the responsibility of the user. Then they can _choose_ whether or not they want to disable the default middleware stack when testing.

```ruby
Shoryuken.configure_server do |config|
  config.server_middleware do |chain|
    ... # configure middleware for default executor
  end

  config.server_middleware.clear if Rails.env.test? # let users decide what do to with default middleware
end
```

Reply:
The other thing to consider, too, is that as written, this change introduces what I think is a breaking API change since if you were using the `InlineExecutor` before, you might have used it in a way that you didn't expect worker-specific middleware to be executed.

One option would be to introduce a new executor (`InlineExecutorWithMiddleware` or something) so that folks can opt-in to the new behavior without a breaking change.